__tests__/app.test.js

```diff
@@ -21,30 +21,63 @@ describe("/api/topics", () => {
       });
   });
 });
-describe('Error handling', () => {
-    test("GET 404: responds with appropriate status and error message for incorrect route", () => {
-        return request(app)
-        .get('/topics')
-        .expect(404)
-        .then(({body: {msg}}) => {
-            expect(msg).toBe('Path not found');
-        })
-    });
+describe("Error handling", () => {
+  test("GET 404: responds with appropriate status and error message for incorrect route", () => {
+    return request(app)
+      .get("/topics")
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Path not found");
+      });
+  });
+});
+describe("/api", () => {
+  test("GET 200: responds an object describing all the available endpoints on the API", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then(({ body: { endpoints } }) => {
+        for (let key in endpoints) {
+          expect(endpoints[key]).toHaveProperty("description");
+          expect(endpoints[key]).toHaveProperty("queries");
+          expect(endpoints[key]).toHaveProperty("exampleResponse");
+          if (["POST", "PATCH", "DELETE"].includes(key)) {
+            expect(endpoints[key]).toHaveProperty("bodyFormat");
+          }
+        }
+      });
+  });
+});
+describe("/api/articles/:article_id", () => {
+  test("GET 200: responds with an article object for the specified article", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then(({ body: { article } }) => {
+        expect(article).toHaveProperty("author");
+        expect(article).toHaveProperty("title");
+        expect(article).toHaveProperty("article_id");
+        expect(article).toHaveProperty("body");
+        expect(article).toHaveProperty("topic");
+        expect(article).toHaveProperty("created_at");
+        expect(article).toHaveProperty("votes");
+        expect(article).toHaveProperty("article_img_url");
+      });
```

Comment:
This test would be good to have a matcher like expect.objectContaining or .toMatchObject to allow for flexibility.